### ISSUE TITLE
Update ReadOnly Field on selection change of select field. 

### ISSUE THREAD

**Issue by codemakeup** [tabalinas/jsgrid#966]
Hi, after looking a lot of your suggestion I have come so far that I am able to update a text field where editing = true when I change the selection on select field. But I want to keep that text field editing = false. 
Below is working code where I want to make "Unit" field as ReadOnly and still able to update from Selection Change of "Item" field . 

----------------------
fields: [
                    {
                        name: "foodItemId", type: "select", valueField: "id", textField: "name", title: "Item", width: "35%", validate: "required", align: "left",
                        items: foodItems,
                        editTemplate: function (value, item) {
                            var grid = this._grid;
                            var $editControl = jsGrid.fields.select.prototype.editTemplate.call(this, value);
                            $editControl.change(function (value, item) {
                                var selectedValue = $(this).val();
                                $.each(foodItemsDataService.foodItems, function (i, item) {
                                    if (item.id == selectedValue) {
                                        grid.fields[1].editControl.val(item.unit.name);
                                        grid.fields[2].editControl.val(item.caloriesPerUnit);
                                    }
                                });
                            });
                            return $editControl;
                        }
                    },
                    {
                        name: "foodItem.unit.name", title: "Unit", type: "text", width: "15%", editing: true
                    }

**Comment by tabalinas** [comment]
Could you provide a jsfiddle to demonstrate the issue?

**Comment by codemakeup** [comment]
I am quite new at JS but I will try to create a Fiddle and share it. In mean time let me explain what i am doing. 
I get my data for the select field from a restful service call and it has three fields. Food item, Unit, and Cals. per Unit. Use has to enter a value for each food item he/she selects. So user is only entering Two fields. Food Item (select field in the grid) and Value (Text Field). The other two (Unit and Cal.s per unit) are part of my object which I get from service. I dont want these fields to be entered by user so I keep it RO. But in doing so when I flip through Food Items Select, these two field dont get updated. If I keep them not RO then and return $editControl as show in my original post, it get updated fine. 

So long story short. When I select an item from Select field I want to update a RO field with some values based on what I have selected. Is it even possible or do I have to keep as Editable fields?

By them I am sorry for the long explanation, i just want make it as clear as possible.

**Comment by codemakeup** [comment]
Here is jsfiddle: [https://jsfiddle.net/ssethi/v8hm8hx7/7/]
On Edit, when i select food item, the calories field which is readonly, should be updated as per what is selected. Its not 100% working fiddle but i hope you will get the idea.

**Comment by tabalinas** [comment]
Checkout the comment to the issue https://github.com/tabalinas/jsgrid/issues/71#issuecomment-129045453.
This should be similar to what you are trying to accomplish.

**Comment by codemakeup** [comment]
Hi, thank you for your help. Yes your are right that it is similar to what I am looking for. But in this fiddle [http://jsfiddle.net/n982uvx1/19/](url), I want to keep the Team column read only and dont want it as Select field. It working perfectly with the selection change but I don't want user be able to change the value which is auto filled by selection in Department Field. Please let me know if I am still not making it clear in what I am looking for.

**Comment by arminus** [comment]
@codemakeup - did you find a solution to your problem? (your fiddles are gone) I'm facing the same issue right now...

In my words: the problem seems to be that there is no editControl if the field is set to editing =  false - so I guess another question would be if editing true|false can by dynamically changed (i.e. true right before the .editControl.val(...) call and back to false right after?

**Comment by codemakeup** [comment]
@arminus  - I didnt find any solution yet. My app is still in dev so I am keeping the fields in Edit mode. If what you are saying works (editing true|false can by dynamically changed ) that could solve the issue too.
